txpool/txpool.go

```diff
@@ -555,38 +566,74 @@ func (t *TxPool) Underpriced(tx *types.Transaction) bool {
 	return underpriced
 }
 
-func (t *TxPool) Discard(slots uint64, force bool) ([]*types.Transaction, bool) {
+func (t *TxPool) Discard(remaining uint64, force bool) ([]*types.Transaction, bool) {
 	dropped := make([]*types.Transaction, 0)
-	for t.remoteTxns.Length() > 0 && slots > 0 {
+	for t.remoteTxns.Length() > 0 && remaining > 0 {
 		tx := t.remoteTxns.Pop()
 		dropped = append(dropped, tx.tx)
 
-		txSlots := numSlots(tx.tx)
-		if slots >= txSlots {
-			slots -= txSlots
+		txSlots := slots(tx.tx)
+		if remaining >= txSlots {
+			remaining -= txSlots
 		} else {
-			slots = 0
+			remaining = 0
 		}
 	}
 
 	// Put back if couldn't make required space
-	if slots > 0 && !force {
+	if remaining > 0 && !force {
 		for _, tx := range dropped {
 			t.remoteTxns.Push(tx)
 		}
 		return nil, false
 	}
+
 	return dropped, true
 }
 
 // increaseSlots increases number of taken slots
 func (t *TxPool) increaseSlots(slots uint64) {
-	atomic.AddUint64(&t.slots, slots)
+	t.gauge.increase(slots)
 }
 
 // increaseSlots decreases number of taken slots
 func (t *TxPool) decreaseSlots(slots uint64) {
```

Comment:
I think this method is completely useless since you can directly call `t.gauge.decrease(slots)`

Reply:
[Removed](https://github.com/0xPolygon/polygon-sdk/pull/213/commits/b0b4046798b2672bb5746693333c71f0f2f7ee49)